From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Thu, 19 Jun 2014 11:37:19 +0200
Subject: [096/249] staging: rtl8723au: rtw_cfg80211_inform_bss(): Report the
 beacon interval already retrieved
Origin: https://git.kernel.org/linus/2620e339e84320814eb9b4c46d9295703ada1010

Use the beacon interval we already retreived in collect_bss_info()
instead of pulling it out of the saved IE array again.

Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/staging/rtl8723au/os_dep/ioctl_cfg80211.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/staging/rtl8723au/os_dep/ioctl_cfg80211.c b/drivers/staging/rtl8723au/os_dep/ioctl_cfg80211.c
index 2621a0e..ff838fc 100644
--- a/drivers/staging/rtl8723au/os_dep/ioctl_cfg80211.c
+++ b/drivers/staging/rtl8723au/os_dep/ioctl_cfg80211.c
@@ -248,7 +248,6 @@ static int rtw_cfg80211_inform_bss(struct rtw_adapter *padapter,
 	u16 channel;
 	u32 freq;
 	u16 notify_capability;
-	u16 notify_interval;
 	u8 *notify_ie;
 	size_t notify_ielen;
 	s32 notify_signal;
@@ -266,9 +265,6 @@ static int rtw_cfg80211_inform_bss(struct rtw_adapter *padapter,
 
 	notify_channel = ieee80211_get_channel(wiphy, freq);
 
-	notify_interval =
-		get_unaligned_le16(
-			rtw_get_beacon_interval23a_from_ie(pnetwork->network.IEs));
 	notify_capability =
 		get_unaligned_le16(
 			rtw_get_capability23a_from_ie(pnetwork->network.IEs));
@@ -290,7 +286,8 @@ static int rtw_cfg80211_inform_bss(struct rtw_adapter *padapter,
 	bss = cfg80211_inform_bss(wiphy, notify_channel,
 				  pnetwork->network.MacAddress,
 				  pnetwork->network.tsf,
-				  notify_capability, notify_interval,
+				  notify_capability,
+				  pnetwork->network.BeaconPeriod,
 				  notify_ie, notify_ielen,
 				  notify_signal, GFP_ATOMIC);
 
